core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -527,7 +536,43 @@ public boolean isTerminated() {
     return terminated;
   }
 
-  /*
+  @Override
+  public ConnectivityState getState(boolean requestConnection) {
+    // TODO: handle requestConnection. At this point, not really sure how.
+    return state;
+  }
+
+  @Override
+  public void notifyWhenStateChanged(ConnectivityState source, Runnable callback) {
```

Comment:
Addressed. PTAL and let me know if the semantics are correct and you had chance to review in general - if so, I can proceed and add tests, docs etc.